File: Assets/LuaFrameworkExtension/UIFramework/Scripts/Core/UIPanelBase.cs
using UnityEngine;
using System.Collections.Generic;

public abstract class UIPanelBase {

    public GameObject gameObject;
    public Transform transform;
    protected UITweener tweener;

    protected virtual void Awake()
    {
    }

    protected virtual void OnCreate(GameObject go)
    {
        gameObject = go;
        transform = gameObject.transform;
    }

    protected virtual void OnDestroy()
    {
    }

    public virtual void Open()
    {
        if (gameObject == null)
        {
            Awake();
        }
        else
        {
            gameObject.SetActive(true);//播放动画的前提是Panel具有可见性
            if (tweener != null) tweener.PlayForward();
        }
    }

    public virtual void Close()
    {
        if (tweener != null) tweener.PlayReverse();
        else gameObject.SetActive(false);
    }

    protected virtual void SetOpenCloseTweener(UITweener tweener, List<EventDelegate> list, EventDelegate.Callback callBack)
    {
        this.tweener = tweener;
        if((list != null) && (callBack != null)) EventDelegate.Add(list, callBack);
    }

    public virtual void Refresh()
    { 
    }

}

File: Assets/Lua/Logic/UIPanelBase.lua
UIPanelBase = {	};

function UIPanelBase:New()
	local o = {};	
	setmetatable(o, UIPanelBase);
	UIPanelBase.__index = UIPanelBase;
	return o;
end

function UIPanelBase:Open()
	if(self.gameObject == nil) then
		self:Awake();
	else
		self.gameObject:SetActive(true);
		if(self.tweener ~= nil) then self.tweener:PlayForward(); end
	end	
end

function UIPanelBase:Close()
	if(self.tweener ~= nil) then self.tweener:PlayReverse(); 
	else self.gameObject:SetActive(false); end
end

function UIPanelBase:SetOpenCloseTweener(tweener, list, callBack)
	self.tweener = tweener;
	if((list ~= nil) and (callBack ~= nil)) then EventDelegate.Add(list, EventDelegate.Callback(callBack)); end
end